### PR TITLE
add api for generating pcm cpd

### PR DESCRIPTION
**Pull request by chehefen**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/mkvgen/src/MkvGenerator.c [hunk 1]
```diff
@@ -1699,6 +1699,42 @@ STATUS getAudioConfigFromAmsAcmCpd(PBYTE pCpd,
...
+
+    blockAlign = channels;
+    bitrate = blockAlign * samplingRate / 8;
+
+    // just in case
+    initializeEndianness();
+
+    MEMSET(buffer, 0x00, KVS_PCM_CPD_SIZE_BYTE);
+
+    putUnalignedInt16LitEndian((PINT16) buffer, format);
```
Comment (MushMal) [inline src/mkvgen/src/MkvGenerator.c]: Perhaps having a PBYTE pCurPtr = buffer then incrementing it by adding SIZEOF(UINT16) is a better choice.

File: src/mkvgen/tst/AudioCpdParserTest.cpp [hunk 2]
```diff
@@ -91,3 +92,34 @@ TEST_F(AudioCpdParserTest, generateAacCpdTest)
...
+TEST_F(AudioCpdParserTest, generatePcmCpdTest)
+{
+    // alaw cpd. sampling rate 44100, channel config is 1. This cpd is generated by gstreamer matroskamux plugin.
+    BYTE cpd[] = {0x06, 0x00, 0x01, 0x00, 0x44, 0xac, 0x00, 0x00, 0x88, 0x15, 0x00, 0x00, 0x01, 0x00, 0x00, 0x00, 0x00, 0x00};
+    const UINT32 samplingFrequency = 44100;
+    const UINT16 channelConfig = 1;
+    BYTE buffer[KVS_PCM_CPD_SIZE_BYTE];
+
+    // invalid format code
+    EXPECT_EQ(STATUS_INVALID_ARG, mkvgenGeneratePcmCpd(0x0001, samplingFrequency, channelConfig, buffer, ARRAY_SIZE(buffer)));
```
Comment (MushMal) [inline src/mkvgen/tst/AudioCpdParserTest.cpp]: Add a test with buffer = NULL, size of buffer = 0, all together

File: src/utils/include/com/amazonaws/kinesis/video/utils/Include.h [hunk 3]
```diff
@@ -877,6 +877,14 @@ extern putUnalignedInt16Func putUnalignedInt16BigEndian;
 extern putUnalignedInt32Func putUnalignedInt32BigEndian;
 extern putUnalignedInt64Func putUnalignedInt64BigEndian;
 
+extern getUnalignedInt16Func getUnalignedInt16LitEndian;
```
Comment (MushMal) [inline src/utils/include/com/amazonaws/kinesis/video/utils/Include.h]: Suggest using full string Little instead of abbreviating Lit

**Comment by alexw91** [comment]
# [Codecov](https://codecov.io/gh/awslabs/amazon-kinesis-video-streams-pic/pull/18?src=pr&el=h1) Report
> Merging [#18](https://codecov.io/gh/awslabs/amazon-kinesis-video-streams-pic/pull/18?src=pr&el=desc) into [master](https://codecov.io/gh/awslabs/amazon-kinesis-video-streams-pic/commit/43cb9df182f42216b71df3d77c53002801f83263&el=desc) will **increase** coverage by `0.04%`.
> The diff coverage is `100.00%`.

[Impacted file tree graph](https://codecov.io/gh/awslabs/amazon-kinesis-video-streams-pic/pull/18?src=pr&el=tree)

```diff
@@            Coverage Diff             @@
##           master      #18      +/-   ##
==========================================
+ Coverage   82.21%   82.26%   +0.04%     
==========================================
  Files          47       47              
  Lines        8272     8292      +20     
==========================================
+ Hits         6801     6821      +20     
  Misses       1471     1471              
```


| [Impacted Files](https://codecov.io/gh/awslabs/amazon-kinesis-video-streams-pic/pull/18?src=pr&el=tree) | Coverage Δ | |
|---|---|---|
| [src/utils/src/Endianness.c](https://codecov.io/gh/awslabs/amazon-kinesis-video-streams-pic/pull/18/diff?src=pr&el=tree#diff-c3JjL3V0aWxzL3NyYy9FbmRpYW5uZXNzLmM=) | `68.14% <ø> (ø)` | |
| [src/mkvgen/src/MkvGenerator.c](https://codecov.io/gh/awslabs/amazon-kinesis-video-streams-pic/pull/18/diff?src=pr&el=tree#diff-c3JjL21rdmdlbi9zcmMvTWt2R2VuZXJhdG9yLmM=) | `91.29% <100.00%> (+0.21%)` | :arrow_up: |

------

[Continue to review full report at Codecov](https://codecov.io/gh/awslabs/amazon-kinesis-video-streams-pic/pull/18?src=pr&el=continue).
> **Legend** - [Click here to learn more](https://docs.codecov.io/docs/codecov-delta)
> `Δ = absolute <relative> (impact)`, `ø = not affected`, `? = missing data`
> Powered by [Codecov](https://codecov.io/gh/awslabs/amazon-kinesis-video-streams-pic/pull/18?src=pr&el=footer). Last update [43cb9df...9652482](https://codecov.io/gh/awslabs/amazon-kinesis-video-streams-pic/pull/18?src=pr&el=lastupdated). Read the [comment docs](https://docs.codecov.io/docs/pull-request-comments).

File: src/mkvgen/src/MkvGenerator.c [hunk 4]
```diff
@@ -1699,6 +1699,48 @@ STATUS getAudioConfigFromAmsAcmCpd(PBYTE pCpd,
...
+STATUS mkvgenGeneratePcmCpd(UINT16 format, UINT32 samplingRate, UINT16 channels, PBYTE buffer, UINT32 bufferLen)
+{
+    STATUS retStatus = STATUS_SUCCESS;
+    UINT16 blockAlign = 0;
+    UINT32 bitrate = 0;
+    PBYTE pCurPtr = NULL;
+
+    CHK(buffer != NULL, STATUS_NULL_ARG);
+    CHK_ERR(bufferLen >= KVS_PCM_CPD_SIZE_BYTE, STATUS_INVALID_ARG, "Buffer is too small");
+    CHK_ERR(format == KVS_PCM_FORMAT_CODE_ALAW || format == KVS_PCM_FORMAT_CODE_MULAW, STATUS_INVALID_ARG,
```
Comment (MushMal) [inline src/mkvgen/src/MkvGenerator.c]: You could define an enum with these values

File: src/mkvgen/src/Include_i.h [hunk 5]
```diff
@@ -164,6 +164,10 @@ extern UINT32 gMkvTrackNameBitsSize;
 // Minimum A_MS/ACM codec private size
 #define MIN_AMS_ACM_CPD_SIZE 16
 
+// Min/Max sampling rate for PCM alaw and mulaw. Referred pad template of gstreamer alawenc plugin
+#define MIN_PCM_SAMPLING_RATE 8000
```
Comment (MushMal) [inline src/mkvgen/src/Include_i.h]: These need to be public as we limit check against them
